gateway/ln-gateway/src/bin/cln_extension.rs

```diff
@@ -496,6 +499,153 @@ impl GatewayLightning for ClnRpcService {
+                        },
+                    )?,
+                    amount: cln_rpc::primitives::AmountOrAll::Amount(
+                        cln_rpc::primitives::Amount::from_sat(request_inner.channel_size_sats),
+                    ),
+                    feerate: None,
+                    announce: None,
+                    minconf: None,
+                    push_msat: Some(cln_rpc::primitives::Amount::from_sat(
+                        request_inner.push_amount_sats,
+                    )),
+                    close_to: None,
+                    request_amt: None,
+                    compact_lease: None,
+                    utxos: None,
+                    mindepth: None,
+                    reserve: None,
+                },
+            ))
+            .await
+            .map_err(|e| {
+                error!("cln fundchannel rpc returned error {:?}", e);
+                tonic::Status::internal(e.to_string())
+            })?;
+
+        Ok(tonic::Response::new(EmptyResponse {}))
+    }
+
+    async fn wait_for_chain_sync(
+        &self,
+        request: tonic::Request<WaitForChainSyncRequest>,
+    ) -> Result<tonic::Response<EmptyResponse>, Status> {
+        let WaitForChainSyncRequest {
+            block_height: target_block_height,
+            max_retries,
+            retry_delay,
+        } = request.into_inner();
+
+        for _ in 0..max_retries {
+            let (blockheight, warning_bitcoind_sync) = self
+                .rpc_client()
+                .await
+                .map_err(|e| Status::internal(e.to_string()))?
+                .call(cln_rpc::Request::Getinfo(
+                    model::requests::GetinfoRequest {},
+                ))
+                .await
+                .map(|response| match response {
+                    cln_rpc::Response::Getinfo(model::responses::GetinfoResponse {
+                        blockheight,
+                        warning_bitcoind_sync,
+                        ..
+                    }) => Ok((blockheight, warning_bitcoind_sync)),
+                    _ => Err(ClnExtensionError::RpcWrongResponse),
+                })
+                .map_err(|e| {
+                    error!("cln getinfo rpc returned error {:?}", e);
+                    tonic::Status::internal(e.to_string())
+                })?
+                .map_err(|e| tonic::Status::internal(e.to_string()))?;
+
+            if blockheight >= target_block_height && warning_bitcoind_sync.is_none() {
+                return Ok(tonic::Response::new(EmptyResponse {}));
+            }
+
+            if let Some(retry_delay) = retry_delay.clone() {
+                sleep(retry_delay.try_into().map_err(|e: DurationError| {
+                    Status::internal(format!(
+                        "Failed to convert prost Duration to std Duration: {}",
+                        e
+                    ))
+                })?)
+                .await;
+            }
+        }
```

Comment:
We do have the [`retry`](https://docs.fedimint.org/fedimint_core/util/fn.retry.html) utility, maybe it can help here too?